apps/web/src/layouts/DaoLayout.tsx

```diff
@@ -20,13 +17,10 @@ function DaoLayout({ children }: { children: ReactNode }) {
 
   const setAddresses = useDaoStore((state) => state.setAddresses)
 
-  const { data } = useContractRead({
-    enabled: !!token,
-    abi: managerAbi,
-    address: PUBLIC_MANAGER_ADDRESS,
-    functionName: 'getAddresses',
-    args: [token as AddressType],
-  })
+  const { data } = useSWR(
+    token ? [SWR_KEYS.DAO_ADDRESSES, token] : null,
+    async () => await getAddresses(token as AddressType)
```

Comment:
do we feel like we need to re-fetch this data client side? I feel like it can just be passed down on the initial page load

Reply:
was thinking it could be relevant for upgrade scenarios but just realized that the addresses will stay static through an upgrade so I can rework to read the props in the layout and remove SWR.

Reply:
sounds good!